src/arrayMethodReduce.js

```diff
@@ -6,6 +6,22 @@
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
     // write code here
+    let startIndex = 0;
+    let accumulator = initialValue;
+
+    if (initialValue === undefined) {
+      accumulator = this[0];
+      startIndex = 1;
+    }
+
+    let currentValue;
```

Comment:
You don't need this variable, you can just use this[i]. By the logic of the code, it is clear that this[i] is the current value